drivers: flash: stm32h5 flash driver reports the actual flash size

FRASTM opened the pull request:
The Max flash size of the stm32H5 serie depends on the mcu device from 256 to 1024 KB in two banks

gautierg-st commented on drivers/flash/flash_stm32l5x.c:
```diff
@@ -21,8 +21,7 @@ LOG_MODULE_REGISTER(LOG_DOMAIN);
 #if defined(CONFIG_SOC_SERIES_STM32H5X)
-/* at this time stm32h5 mcus have 128KB (stm32h50x) or 2MB (stm32h56x/57x) */
-#define STM32_SERIES_MAX_FLASH	2048
+#define STM32_SERIES_MAX_FLASH	(CONFIG_FLASH_SIZE)
```
You could group H5 with U5 below to avoid duplication

FRASTM replied:
yes expect that comment does not apply for stm32H5

erwango replied:
According to the comment, defining `STM32_SERIES_MAX_FLASH as `CONFIG_FLASH_SIZE` avoid the unexpected check (of 2 bank discontinuity case)
So either this is also correct for H5 and then comments also applies, either the fix is not correct.

kartben commented:
ping @erwango / ST team :)

kartben commented:
@FRASTM new git setup? :) you have a compliance error to fix

FRASTM commented:
> @FRASTM new git setup? :) you have a compliance error to fix

I do not see where this comes from !?

FRASTM committed:
drivers: flash: stm32h5 flash driver reports the actual flash size

The Max flash size of the stm32H5 serie depends on the mcu device
from 256 to 1024 KB, in two banks.

Signed-off-by: Francois Ramu <francois.ramu@st.com>
